src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -43,31 +40,65 @@ public void TexlDateOverloads_Negative(string script)
         [InlineData("Date(2000,1,1) - Date(1999,1,1)", "n")]
         [InlineData("Time(2,1,1) - Time(2,1,1)", "n")]
         [InlineData("5 - DateTimeValue(\"1 Jan 2015\")", "d")]
-        [InlineData("5 - Date(2000,1,1)", "d")]
         [InlineData("5 - Time(2000,1,1)", "T")]
         [InlineData("-Date(2001,1,1)", "D")]
         [InlineData("-Time(2,1,1)", "T")]
         [InlineData("-DateTimeValue(\"1 Jan 2015\")", "d")]
+        [InlineData("Date(2000,1,1) + 5", "d")]
+        [InlineData("5 + Date(2000,1,1)", "d")]
+        [InlineData("5 - Date(2000,1,1)", "d")]
         [InlineData("Time(20,1,1) + Time(19,1,1)", "T")]
         [InlineData("DateTimeValue(\"1 Jan 2015\") + Time(20,1,1)", "d")]
         [InlineData("Time(20,1,1) + DateTimeValue(\"1 Jan 2015\")", "d")]
         [InlineData("DateTimeValue(\"1 Jan 2015\") - Time(20,1,1)", "d")]
         [InlineData("DateValue(\"1 Jan 2015\") - Time(20,1,1)", "d")]
-        public void TexlDateOverloads(string expression, string expectedType)
+        public void TexlDateOverloads(string script, string expectedType)
         {
-            var engine = new Engine(new PowerFxConfig());
-            var result = engine.Check(expression);
+            Assert.True(DType.TryParse(expectedType, out var type), script);
+            Assert.True(type.IsValid, script);
 
-            Assert.True(DType.TryParse(expectedType, out var expectedDType));
-            Assert.Equal(expectedDType, result._binding.ResultType);
-            Assert.True(result.IsSuccess);
+            TestSimpleBindingSuccess(script, TestUtils.DT(expectedType));
+        }
+
+        [Theory]
+        [InlineData("DateTimeValue(\"1 Jan 2015\") + DateTimeValue(\"1 Jan 2015\")")]
+        [InlineData("DateTimeValue(\"1 Jan 2015\") + Date(2000,1,1)")]
+        [InlineData("Date(2000,1,1) + Date(1999,1,1)")]
+        [InlineData("Date(2000,1,1) + DateTimeValue(\"1 Jan 2015\")")]
+        [InlineData("Time(20,1,1) - DateTimeValue(\"1 Jan 2015\")")]
+        [InlineData("Time(20,1,1) - Date(2000,1,1)")]
+        public void TexlDateOverloads_Negative(string script)
+        {
+            // TestBindingErrors(script, DType.Error);
+            var engine = new Engine(new PowerFxConfig());
+            var result = engine.Check(script);
+            
+            Assert.Equal(DType.Error, result._binding.ResultType);            
+            Assert.False(result.IsSuccess);
         }
 
         [Theory]
         [InlineData("DateAdd([Date(2000,1,1)],1)", "*[Value:d]")]
         [InlineData("DateAdd([Date(2000,1,1)],[3])", "*[Value:d]")]
         [InlineData("DateAdd(Table({a:Date(2000,1,1)}),[3])", "*[a:d]")]
         [InlineData("DateAdd(Date(2000,1,1),[1])", "*[Result:d]")]
+        [InlineData("DateAdd(DateTimeValue(\"1 Jan 2015\"), 2)", "d")]
+        [InlineData("DateAdd(DateTimeValue(\"1 Jan 2015\"), 2, TimeUnit!Years)", "d")]
```

Comment:
>! [](http://example.com/codeflow?start=71&length=1)

Even though this is supported (`!` as the field selector), let's not have tests showing this. We want the tests to be used as a "template" for people to see how the language can be used.

Reply:
Although ... we're still not allowed to break it, so we still need at least one test for it.  This PR is capturing existing tests that we have in PA client and run closer to fx core.

Reply:
We can have one or a few _specific_ tests that use this syntax for back-compat purposes, but it shouldn't be sprinkled throughout the codebase.

Reply:
removed tests with field selector, while keeping just few.